onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -321,6 +324,11 @@ def main():
 
     _setup_logger(args.verbose)
 
+    if os.path.realpath(args.input) == os.path.realpath(args.output):
```

Comment:
Also need to add checking dirname is the same when args.use_external_data_format is true? If input and output uses the same path, data will be overwritten anyway. And what do you think if existing instead of warning? Do you think we have any case of using the same input and output name?

Reply:
hmm..I think this line is enough for check paths for whatever the args.use_external_data_format is if we were to use this file separately?
I am not quite following the second question is that saying we should like return the program if input and output share the same path? As warning user may loss the original model...
From our side, no, we will output it to a different path.